### PR TITLE
[ci] add GitHub actions FPGA job

### PR DESCRIPTION
**Pull request by nbdd0121**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -299,6 +299,91 @@ jobs:
           gcloud storage cp latest.txt $BUCKET_URI/latest.txt
           gcloud storage cp $BUCKET_URI/bitstream-${{ github.sha }}.tar.gz $BUCKET_URI/bitstream-latest.tar.gz
 
+  # CW310 FPGA jobs.
+  execute_test_rom_fpga_tests_cw310:
+    name: CW310 Test ROM Tests
+    needs: [chip_earlgrey_cw310, sw_build_test]
```
Comment (jwnrt) [inline .github/workflows/ci.yml:305]: I think we discussed dropping the `sw_build_test` dependency, did we agree to keep it?

File: azure-pipelines.yml
Comment (jwnrt) [inline azure-pipelines.yml:1]: I think you can remove the CW310 bitstream build from Azure now, right?

Comment (a-will) [inline azure-pipelines.yml:1]: Shouldn't chip_earlgrey_cw310 be removed as well? It seems we're running duplicate CW310 build jobs right now.

Comment (nbdd0121) [reply azure-pipelines.yml:1]: It's currently a dependency of the "Package & release" step. Although I think the artifact generated by that step is not being used anymore and we can probably remove it, I'll do it as a follow-up PR.

**Review by jwnrt — APPROVED** [review]
Updated version LGTM

**Pull request merged**